core/src/main/java/com/linecorp/armeria/internal/client/DefaultClientRequestContext.java

```diff
@@ -849,4 +850,28 @@ private String toStringSlow(@Nullable Channel ch, @Nullable RequestLogAccess par
             return buf.toString();
         }
     }
+
+    @Override
+    public CompletableFuture<Void> initiateConnectionShutdown() {
+        final CompletableFuture<Void> completableFuture = new CompletableFuture<>();
+
+        setAdditionalRequestHeader(HttpHeaderNames.CONNECTION, HttpHeaderValues.CLOSE);
+        log().whenRequestComplete().whenComplete((ignore, ex) -> {
```

Comment:
Could use `handle` instead because `CompletableFuture.whenComplete()` is known to have performance overhead in old JDK versions.